### ISSUE TITLE
Username could not be found

### ISSUE THREAD

**Issue by z3ky** [tchapi/davis#125]
I'm trying to run davis with the provided docker-compose.yml.

After the first start I ran `bin/console doctrine:migrations:migrate` from within the container, logged in with the standard admin account and created some users.
When I try to login with that users, I get: Username could not be found

davis/var/log/prod.log gives no information

**Comment by tchapi** [comment]
Hi @z3ky 👋🏼 

Where are you trying to login? To the admin dashboard? There is only one admin account, and the users that you created are DAV users (who can't login to the admin dashboard) - maybe there was a confusion here?

**Comment by z3ky** [comment]
Hi,
yes, to the admin dashboard. Ok I understand the different kind of users.

But, login with macos calendar app also gives me wrong user or password.
I tried /dav, /dav/principals and /dav/principals/username as endpoints. Is one of them correct?

**Comment by tiiiecherle** [comment]
I use davis with macOS clients for calendars, reminders and address book.

To add an account on macOS 
got to system preferences
internet accounts
add account
add other account
Caldav or Carddav
account type: manual
user name: davis user name
password: davis user password
server: URL-TO-YOUR-SERVER.COM:443

Make sure your webserver is configured correctly according to the davis [readme](https://github.com/tchapi/davis#-access--webserver) and that redirects are set correctly, too.

Hope this helps.

**Comment by z3ky** [comment]
Thanks @tiiiecherle that helped.

The problem is the docker deployment. For nginx docker binds the host port 9000 and port 80 internally. But, the rewrite rules don't respect port 9000 and redirect to 80.
I changed the rules to `rewrite ^/.well-known/carddav http://nginx.server.com:9000/dav/ redirect;`

But the macOS calendar app still doesn't, even if I add the account using "Extended" and select to correct port and disable ssl. It only works when ssl in nginx is enabled.

To sum up: the rewrite rules in the docker deployment needs the docker host port for nginx and macOS calendar app needs ssl. My rewrite rules are

```
rewrite ^/.well-known/caldav https://nginx.server.com:9000/dav/ redirect;
rewrite ^/.well-known/carddav https://nginx.server.com:9000/dav/ redirect;
```

**Comment by tiiiecherle** [comment]
I only use and only tried https and not http.
For me the example nginx configuration from the [readme](https://github.com/tchapi/davis#-access--webserver) works perfectly.
I use davis with a local installation without docker.

**Comment by holow29** [comment]
> To sum up: the rewrite rules in the docker deployment needs the docker host port for nginx and macOS calendar app needs ssl.
IIRC MacOS and iOS require SSL. They won't work otherwise - this is a limitation of the clients on those platforms.

**Comment by tchapi** [comment]
@z3ky are you good with the proposed circumvention? Do you need further help?

**Comment by z3ky** [comment]
yes, works for me.
Maybe the need to edit the rewrite rules for .well-known when using docker deployment could be added to the readme.

**Comment by tchapi** [comment]
> Maybe the need to edit the rewrite rules for .well-known when using docker deployment could be added to the readme.

I personally use the rules that I put in the README, for a Docker deployment (one container for Davis: `davis`, one for nginx). Here is my simplified `nginx.conf` file:

```
http {

    upstream docker-davis {
        server davis:9000;
    }

    server {
        listen 80 default;
        listen [::]:80 default;

        root /var/www/davis/public/;
        index index.php;

        rewrite ^/.well-known/caldav /dav/ redirect;
        rewrite ^/.well-known/carddav /dav/ redirect;

        location / {
            try_files $uri $uri/ /index.php$is_args$args;
        }

        location ~ ^(.+\.php)(.*)$ {
            try_files       $fastcgi_script_name =404;
            include         fastcgi_params;
            fastcgi_pass    docker-davis;
            fastcgi_param   SCRIPT_FILENAME  $document_root$fastcgi_script_name;
            fastcgi_param   PATH_INFO        $fastcgi_path_info;
            fastcgi_split_path_info  ^(.+\.php)(.*)$;
        }
    }
}
```

So I'm not sure what to change in the readme in fact. You don't need to add the port if the redirect is local, as it happens before touching your upstream.

**Comment by extrafein** [comment]
> Hi, yes, to the admin dashboard. Ok I understand the different kind of users.
> 
> But, login with macos calendar app also gives me wrong user or password. I tried /dav, /dav/principals and /dav/principals/username as endpoints. Is one of them correct?

Just to give you a hint; this URL is working for me for MacOS and iOS:
https://<server>/dav/principals/<username>/
